src/colmap/scene/frame.h

```diff
@@ -114,10 +91,12 @@ class Frame {
   // case, where rig modeling is no longer needed.
   std::optional<Rigid3d> frame_from_world_;
 
-  // [Optional] Rig calibration
-  std::shared_ptr<class Rig> rig_ = nullptr;
+  // [Optional] Rig calibration.
+  const class Rig* rig_ = nullptr;
```

Comment:
I made this change for consistency with the Image::CameraPtr. In a follow-up, I will homogeneously use shared_ptr's for the CameraPtr/FramePtr/RigPtr, which requires us to store them as shared_ptr's in the reconstruction object.